ajax/impact.php

```diff
@@ -47,41 +47,67 @@
 switch ($_SERVER['REQUEST_METHOD']) {
    // GET request: build the impact graph for a given asset
    case 'GET':
-      $itemtype =  $_GET["itemtype"]  ?? "";
-      $items_id =  $_GET["items_id"]  ?? "";
-      $view     =  $_GET["view"]      ?? "graph";
-
-      // Check required params
-      if (empty($itemtype) || empty($items_id)) {
-         Toolbox::throwError(400, "Missing itemtype or items_id");
-      }
+      $action = $_GET["action"]  ?? "";
+
+      switch ($action) {
+         case "search":
+            $itemtype = $_GET["itemtype"] ?? "";
+            $used     = $_GET["used"]     ?? "[]";
+            $filter   = $_GET["filter"]   ?? "";
+            $page     = $_GET["page"]     ?? 0;
+
+            // Check required params
+            if (empty($itemtype)) {
+               Toolbox::throwError(400, "Missing itemtype");
```

Comment:
Not related to current PR, but I think the name of the method is not correct, since it only handle JSON related (also this method miss phpdoc)

Reply:
I've added phpdoc and an additional param to specify the format (default is json).